src/components/views/room_settings/AliasSettings.js

```diff
@@ -262,6 +262,7 @@ module.exports = React.createClass({
                     items={this.state.domainToAliases[localDomain] || []}
                     newItem={this.state.newAlias}
                     onNewItemChanged={this.onNewAliasChanged}
+                    canEdit={this.props.canSetAliases}
```

Comment:
This seems like it ends up preventing you from adding aliases local to your homeserver. The flag given to the class is for canonical aliases, which should certainly be restricted. However arbitrary aliases are different and bypass power levels: https://github.com/matrix-org/synapse/issues/2402

Reply:
I'm not convinced this is relevant here. `canSetAliases` is always `true` apparently, so this is more for clarity. `canSetCanonicalAlias` is for canonical aliases.

Reply:
Yeah, the current situation is that we hardcode `canSetAliases` to `true` but then also never actually read it anywhere in `AliasSettings`, right? We may as well either wire the thing up if it looks trivial enough to do so (which it looks like it is) or remove it.